src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/gadgets/MultiTool.java

```diff
@@ -66,13 +66,10 @@ protected ItemUseHandler getItemUseHandler() {
 
             int index = selectedMode.getOrDefault(p.getUniqueId(), 0);
 
-            if (!p.isSneaking()) {
-                if (removeItemCharge(item, COST)) {
-                    SlimefunItem sfItem = modes.get(index).getItem();
-
-                    if (sfItem != null) {
-                        sfItem.callItemHandler(ItemUseHandler.class, handler -> handler.onRightClick(e));
-                    }
+            if (!p.isSneaking() && removeItemCharge(item, COST)) {
+                SlimefunItem sfItem = modes.get(index).getItem();
+                if (sfItem != null) {
+                    sfItem.callItemHandler(ItemUseHandler.class, handler -> handler.onRightClick(e));
                 }
```

Comment:
Now you changed the logic of this...
`removeItemCharge` needs to be in a seperate if, otherwise the mode-changing else block will run if you are out of energy.